lib/persistent.js

```diff
@@ -31,6 +35,53 @@ module.exports = class PersistentNode {
     return this.dht.id.equals(TMP)
   }
 
+  onget (req, { mutable }) {
+    if (!req.target) return req.reply(null)
+    const store = mutable ? this.mutables : this.immutables
+    if (mutable) {
+      const k = req.target.toString('hex')
+      const value = store.get(k)
+      if (!value) return req.reply(Buffer.from('')) // empty response so initial puts can still find nodes
+      return req.reply(value)
+    }
+
+    const { salt, seq } = cenc.decode(messages.mutableReq, req.value)
+    const k = salt ? req.target.toString('hex') + salt.toString('hex') : req.target.toString('hex')
+    const value = store.get(k)
+    if (!value) return req.reply(Buffer.from('')) // empty response so initial puts can still find nodes
+    const local = cenc.decode(messages.mutableRes, value)
+    if (local.seq < seq) return req.reply(null)
+    return req.reply(value)
+  }
+
+  onput (req, { mutable }) {
+    if (!req.target || !req.value) return req.reply(null)
```

Comment:
also add !req.token

Reply:
In general, whenever the requester does something wrong just return instead of reply(null) - they are not following the protocol, they don't reserve a reply